src/main/kotlin/com/autonomousapps/model/declaration/Variant.kt

```diff
@@ -1,35 +1,49 @@
 package com.autonomousapps.model.declaration
 
+import com.autonomousapps.internal.utils.capitalizeSafely
+import org.gradle.api.tasks.SourceSet
+
 /**
  * A "Variant" has two meanings depending on context:
  * 1. For the JVM, it is simply the source set (e.g., "main" and "test").
- * 2. For Android, it is the combination of _variant_ (e.g., "debug" and "release") and [SourceSetKind] ("main" and
+ * 2. For Android, it is the combination of _variant_ (e.g., "debug" and "release") and [SourceSet] ("main" and
  * "test").
  */
 data class Variant(
-  val variant: String,
-  val kind: SourceSetKind
+  val androidVariant: String,
+  val sourceSetName: String
```

Comment:
This needs more clarification. I think if this class would get the two aspects clearly separated:
- sourceSetName
- androidVariantName (buildType+flavor)

And then this class could encapsulate the knowledge how these information is combined into a complete "Variant Name" used for Buckets and Tasks (see also `variantSpecificBucketName()` below). I think then the code using this could become cleaner. 

Right now there seems to be a bit of a mixup between when something is just a 'Source Set' and something is an 'Android Variant'.